lib/custom_wizard/submission.rb

```diff
@@ -96,6 +97,27 @@ def self.get(wizard, user_id)
     new(wizard, data, user_id)
   end
 
+  def self.cleanup_incomplete_submissions(wizard, user_id = nil)
+    user_id = user_id || wizard.user.id
+    all_submissions = list(wizard, user_id: user_id)
+    incomplete_submissions = all_submissions.select { |submission| !submission.submitted_at }
+    return if incomplete_submissions.length < 2
+
```

Comment:
@fzngagan The logic in this method works, but it feels like it could use a bit more work. This could be reduced to something like this (pseudo code)

```
has_unfinished = false

valid_data = list(wizard, user_id: user_id)
 .sort_by(submitted_at, updated_at) // sort by submitted_at, then updated_at, see further: https://stackoverflow.com/questions/4309723/ruby-sort-by-multiple-values
 .reduce([]) do |result, submission|
    if submission.submitted_at || !has_unfinished
      result.push(submission)
      has_unfinished if !submission.submitted_at
    end

    result
 end.map { |submission| submission.data_to_save(submission) }

PluginStore.set("#{wizard.id}_#{KEY}", user_id, valid_data)
```

Reply:
A few commits back, I had written the logic to do exactly that, while also accomodating the data without a `updated_at` timestamp i.e. the existing data.
```
incomplete_submissions.sort! do |a, b|
      if a.updated_at.present? && b.updated_at.present?
        Time.iso8601(b.updated_at) <=> Time.iso8601(a.updated_at)
      elsif a.updated_at.present?
        -1
      else
        1
      end
    end
```

Push all the submissions with `submitted_at` to the result. Now the remaining submissions are sorted by the most recent incomplete submission. So pick only the first one of them. Makes a lot of sense and gets rid of a lot of redundant logic. Thanks